src/pages/index.js

```diff
@@ -65,55 +42,81 @@ function handleEditProfileButtonClick() {
   formValidators["profile-form"].disableSubmitButton();
 }
 
-profilePopup.setEventListeners();
-
-editProfileButton.addEventListener("click", handleEditProfileButtonClick);
+function handleDelButtonClick(card) {
+  deleteDialogue.open();
+  deleteIdentification = card;
+}
 
-function renderCard(card) {//to address the recursive code I had been writing in the section and addphoto initializations
-  const newCard = new Card(card, "#card", handlePhotoClick);
+function renderCard(card) {
+  const newCard = new Card(card, "#card", handlePhotoClick, handleDelButtonClick, handleLikeClick);//ammended the delete and like handler so it could tell the other popup to open when the Card's del button was pressed and utilize the api.  
   const newCardElement = newCard.generateCardElement();
   cardSection.addItem(newCardElement);
 }
 
-const addPopup = new PopupWithForm(addPhotoPopup, {
-  handleFormSubmit: ({ title, image_url: imageUrl }) => {
-    const newObj = {
-      name: title,
-      link: imageUrl
-    };
-    renderCard(newObj);    
-    formValidators["add-photo-form"].resetFormValidation();
-    addPopup.close();
-  },
-});
-
-
-addPopup.setEventListeners();
-
-const cardSection = new Section(
-{
-  items: initialCards,
-  renderer: renderCard,
-},
-".cards"
-);
-
-cardSection.renderItems();
+function handleDeleteSubmit() {
+  deleteDialogue.renderLoading(true, "Deleting...");
+  api.deleteCard(deleteIdentification.id)
+  .then(() => {
+    deleteIdentification.deleteElement();
+    //formValidators["delete-photo-form"].disableSubmitButton(); //I think commenting this out might be a mistake- so I'm just noting it out, not deleting
+  })
+  .then(() => {    
+      deleteDialogue.close();
+      deleteIdentification = "";
+  })
+  .catch((err) => {
+    console.log(err);
+    deleteDialogue.renderLoading(true, "Error while deleting! Please wait and try again!");
```

Comment:
`    deleteDialogue.renderLoading(true, "Error while deleting! Please wait and try again!");
`

This is not need in `catch`, because you stop the loading effect in `finally` (it 100% changes the button text anyway)